gitpod/Dockerfile

```diff
@@ -0,0 +1,104 @@
+#
+# Dockerfile for pandas development
+#
+# Usage:
+# -------
+#
+# To make a local build of the container, from the 'Docker-dev' directory:
+# docker build  --rm -f "Dockerfile" -t <build-tag> "."
+#
+# To use the container use the following command. It assumes that you are in
+# the root folder of the pandas git repository, making it available as
+# /home/pandas in the container. Whatever changes you make to that directory
+# are visible in the host and container.
+# The docker image is retrieved from the pandas dockerhub repository
+#
+# docker run --rm -it -v $(pwd):/home/pandas pandas/pandas-dev:<image-tag>
+#
+# By default the container will activate the conda environment pandas-dev
+# which contains all the dependencies needed for pandas development
+#
+# To build and install pandas run:
+#               python setup.py build_ext -j 4
+#               python -m pip install -e . --no-build-isolation
+#
+# This image is based on: Ubuntu 20.04 (focal)
+# https://hub.docker.com/_/ubuntu/?tab=tags&name=focal
+# OS/ARCH: linux/amd64
+FROM gitpod/workspace-base:latest
+
+ARG MAMBAFORGE_VERSION="4.11.0-0"
+ARG CONDA_ENV=pandas-dev
+ARG PANDAS_HOME="/home/pandas"
+
+
+# ---- Configure environment ----
+ENV CONDA_DIR=/home/gitpod/mambaforge3 \
+    SHELL=/bin/bash
+ENV PATH=${CONDA_DIR}/bin:$PATH \
+    WORKSPACE=/workspace/pandas
+
+
+# -----------------------------------------------------------------------------
+# ---- Creating as root - note: make sure to change to gitpod in the end ----
+USER root
+
+# Avoid warnings by switching to noninteractive
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Configure apt and install packages
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends apt-utils dialog 2>&1 \
+    #
+    # Install tzdata and configure timezone (fix for tests which try to read from "/etc/localtime")
+    && apt-get -y install tzdata \
+    && ln -fs /usr/share/zoneinfo/Etc/UTC /etc/localtime \
+    && dpkg-reconfigure -f noninteractive tzdata \
+    #
+    # Verify git, process tools, lsb-release (common in install instructions for CLIs) installed
+    && apt-get -y install git iproute2 procps iproute2 lsb-release \
+    #
+    # cleanup
+    && apt-get autoremove -y \
+    && apt-get clean -y \
+    && rm -rf /var/lib/apt/lists/*
+
+# Switch back to dialog for any ad-hoc use of apt-get
+ENV DEBIAN_FRONTEND=dialog
+
+# Allows this Dockerfile to activate conda environments
+SHELL ["/bin/bash", "--login", "-o", "pipefail", "-c"]
+
+# -----------------------------------------------------------------------------
+# ---- Installing mamba  ----
+RUN wget -q -O mambaforge3.sh \
```

Comment:
Do we even need to use conda / mamba within Docker? Wondering if it’s worth having an extra layer of virtualization versus simplification not going through all these steps

Reply:
(Maybe related), if we are not expecting users to be able to alter the dev environment when using `gitpod`, maybe mamba/conda can be removed entirely? https://pythonspeed.com/articles/conda-docker-image-size/

Reply:
If we want to keep using conda / mamba, another option could also be to use micromamba, which avoids having an additional base environment (the second point mentioned in the linked blog post for why the resulting docker images become big)

Reply:
I was thinking we could simply pip install from requirements.txt

Reply:
> FWIW I put some comparisons on the mamba within Docker versus without in https://github.com/pandas-dev/pandas/pull/49981

Thanks, that's useful. I will comment about the usage of mamba there further.